pkg/repository/local_git.go

```diff
@@ -295,7 +300,90 @@ func (lg *LocalGit) GetDirContentSHA(ctx context.Context, gitPath, dirPath strin
 		}
 		return "", errLocalGitCannotGetStatusCannotRecover
 	}
-	return string(output), nil
+
+	// Create a git ls-files command to list all files and pipe it into a bufio.Scanner
+	// to filter out ignored files
+	lsCmd := lg.exec.CreateCommand(ctx, gitPath, lg.gitPath, "--no-optional-locks", "ls-files", "-s", dirPath)
+	pr1, pw1 := io.Pipe()
+	var pe1 bytes.Buffer
+
+	defer pw1.Close()
+	defer pr1.Close()
+
+	lsCmd.Stdout = pw1
+	lsCmd.Stderr = &pe1
+
+	if err := lsCmd.Start(); err != nil {
+		return handleError(err)
+	}
+
+	hashCmd := lg.exec.CreateCommand(ctx, gitPath, lg.gitPath, "--no-optional-locks", "hash-object", "--stdin")
+	scanner := bufio.NewScanner(pr1)
+
+	pr2, pw2 := io.Pipe()
+	var pe2 bytes.Buffer
+
+	defer pw2.Close()
+	defer pr2.Close()
+
+	if err := hashCmd.Start(); err != nil {
+		return handleError(err)
+	}
+
+	var result bytes.Buffer
+
+	hashCmd.Stdin = pr2
+	hashCmd.Stdout = &result
+	hashCmd.Stderr = &pe2
+
+	go func() {
+		lsFilesColumnCount := 4
+
+		for scanner.Scan() {
+			line := scanner.Text()
+			parts := strings.Fields(line)
+
+			if len(parts) < lsFilesColumnCount {
+				// if we can't parse a line, add the file to the list.
+				// It's better to over build than to over cache
+				//nolint:errcheck
+				pw2.Write([]byte(line + "\n"))
+			} else {
+				filename := parts[3]
+				shouldIgnore, err := lg.ignorer.Ignore(filename)
```

Comment:
Adding this within `GetDirContentSHA` might provoke that all images are rebuilt, as the sha would be different even if there are no changes. 

I think that changing it only for `Diff` function and `ListUntrackedFiles` would be enough. Do you recall which case we wanted to cover adding this?

The relevant part would be not to trigger builds when there is a change in an ignored file, but this function I think it is not being used for  that.

Reply:
Talked offline about this. This is needed because the output of `git ls` command would be different if there is a change in a file, so it would trigger a new rebuild. So, even if the first build is retriggered, it is fine.

We have talked also about possible performance implications or error. That's why we will add an environment variable to enable this new behavior only when it is enabled, so we can test and tell customers to enable it